pkg/vault/contracts/authorizer/TimelockAuthorizerMigrator.sol

```diff
@@ -61,27 +70,41 @@ contract TimelockAuthorizerMigrator {
         vault = _vault;
 
         for (uint256 i = 0; i < _rolesData.length; i++) {
-            rolesData.push(OldRoleData(_rolesData[i].role, _rolesData[i].target));
+            RoleData memory roleData = _rolesData[i];
+            // We require that any permissions being copied from the old Authorizer must exist on the old Authorizer.
+            // This simplifies verification of the permissions being added to the new TimelockAuthorizer.
+            require(_oldAuthorizer.canPerform(roleData.role, roleData.grantee, roleData.target), "UNEXPECTED_ROLE");
+            rolesData.push(roleData);
+        }
+        for (uint256 i = 0; i < _grantersData.length; i++) {
+            // There's no concept of a "granter" on the old Authorizer so we cannot verify these onchain.
+            // We must manually verify that these permissions are set sensibly.
+            grantersData.push(RoleData(_grantersData[i].grantee, _grantersData[i].role, _grantersData[i].target));
+        }
+        for (uint256 i = 0; i < _revokersData.length; i++) {
+            // Similarly to granters, we must manually verify that these permissions are set sensibly.
+            revokersData.push(RoleData(_revokersData[i].grantee, _revokersData[i].role, _revokersData[i].target));
         }
 
         // Enqueue a root change execution in the new authorizer to set it to the desired root address
         rootChangeExecutionId = _newAuthorizer.scheduleRootChange(_root, _arr(address(this)));
     }
 
     /**
-     * @dev Tells whether the migration has been completed or not
+     * @notice Returns whether the migration has been completed or not
      */
     function isComplete() public view returns (bool) {
-        return migratedRoles >= rolesData.length;
+        // As we set up the revoker roles last we can use them to determine whether the full migration is complete.
+        return revokersMigrated >= revokersData.length;
     }
 
     /**
      * @dev Migrate roles from the old authorizer to the new one
-     * @param n Number of roles to migrate, use 0 to migrate all the remaining ones
+     * @param n Number of roles to migrate, use MAX_UINT256 to migrate all the remaining ones
      */
     function migrate(uint256 n) external {
         require(!isComplete(), "MIGRATION_COMPLETE");
```

Comment:
The requires that check the completion status could be made into modifiers. Even though they're local to this contract, it would be a bit more readable.

`function migrate(uint256 n) external onlyBeforeComplete {...}`

Reply:
We only make these checks on two functions I don't think modifiers give us much benefits here as opposed to just inlining the check.